test/blackbox-tests/test-cases/pkg/dev-tool-conflict-test.t

```diff
@@ -0,0 +1,253 @@
+
+The scenario here is, the fake ocamlforamt dev-tool depends on
+printer.1.0 and the project depends on a different version, printer.2.0.
+It shows those 2 does not conflicts and the dev tools dependencies do not leak
+into the user build environment.
+
+  $ . ./helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat:
+  $ mkdir ocamlformat
+  $ cd ocamlformat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name ocamlformat))
+  > EOF
+
+  $ cat > ocamlformat.ml <<EOF
+  > let () = Printer.print ()
+  > EOF
+
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat)
+  >  (libraries printer))
+  > EOF
+
+  $ cd ..
+  $ tar -czf ocamlformat.tar.gz ocamlformat
+  $ rm -rf ocamlformat
+
+Make a printer lib(version 1) that prints "formatted":
+  $ mkdir printer
+  $ cd printer
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name printer))
+  > EOF
+
+  $ cat > printer.ml <<EOF
+  > let print () = print_endline "formatted"
+  > EOF
+
+  $ cat > dune <<EOF
+  > (library
+  >  (public_name printer))
+  > EOF
+
+  $ cd ..
+  $ tar -czf printer.1.tar.gz printer
+
+Make a printer lib(version 2) that prints "Hello world":
+  $ cd printer
+  $ cat > printer.ml <<EOF
+  > let print () = print_endline "Hello world!"
+  > EOF
+  $ cd ..
+  $ tar -czf printer.2.tar.gz printer
+  $ rm -rf printer
+
+A printer 1.0 into opam-repository
+  $ mkpkg printer 1.0 <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/printer.1.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum printer.1.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Add printer.2.0 into opam-repository
+  $ mkpkg printer 2.0 <<EOF
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/printer.2.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum printer.2.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a package for the fake ocamlformat library which depends on printer.1.0:
+  $ mkpkg ocamlformat 0.26.2 <<EOF
+  > depends: [
+  >  "printer" {= "1.0"}
+  > ]
+  > build: [
+  >   [
+  >     "dune"
+  >     "build"
+  >     "-p"
+  >     name
+  >     "@install"
+  >   ]
+  > ]
+  > url {
+  >  src: "file://$PWD/ocamlformat.tar.gz"
+  >  checksum: [
+  >   "md5=$(md5sum ocamlformat.tar.gz | cut -f1 -d' ')"
+  >  ]
+  > }
+  > EOF
+
+Make a project that uses the fake ocamlformat:
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package
+  >  (name foo)
+  >  (depends (printer (= 2.0))))
+  > EOF
+
+  $ cat > foo.ml <<EOF
+  > let () = Printer.print ()
+  > EOF
+
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name foo)
+  >  (libraries printer))
+  > EOF
+
+  $ cat > .ocamlformat <<EOF
+  > version = 0.26.2
+  > EOF
+
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.13)
+  > (lock_dir
+  >  (path "dev-tools.locks/ocamlformat")
```

Comment:
It seems weird to have this file in the source. Is it expected to be committed?

Reply:
This prevent solving with the default `opam-repository`.

Reply:
Maybe it does, but my question is whether you expect your `dev-tools.locks` file to be committed?

Reply:
It's a directory, yes it is committed for this PR, the aim was to put it into the `_build` directory but at the moment dune only load locked directories(dune.lock) from the source. 

We would like make dune to load also from the  build in another PR, to be properly done and to not postpone this PR because of that.